src/hooks/useServerUpdateRequiredMonitor.ts

```diff
@@ -23,27 +24,17 @@ const useServerUpdateRequiredMonitor = (draftSpecs: DraftSpecQuery[]) => {
                 return Object.entries(bindings).some(
                     ([collection, bindingUUIDs]) => {
                         return bindingUUIDs.some((bindingUUID) => {
-                            const targetBindingIndex = Object.hasOwn(
-                                bindings,
-                                collection
-                            )
-                                ? bindings[collection].findIndex(
-                                      (uuid) => uuid === bindingUUID
-                                  )
-                                : -1;
+                            const expectedBindingIndex =
+                                resourceConfigs[bindingUUID].meta.bindingIndex;
```

Comment:
If `resourceConfigs[bindingUUID]` is guaranteed to exist then down below we should remove the optional chaining down below.